Sources/MongoSwift/MongoCollection+FindAndModify.swift

```diff
@@ -0,0 +1,140 @@
+import libmongoc
+
+/// An extension of `MongoCollection` encapsulating find and modify operations.
+extension MongoCollection {
```

Comment:
note that I have modified all the find and modify commands to support using the `MongoCollection`'s `CollectionType`.